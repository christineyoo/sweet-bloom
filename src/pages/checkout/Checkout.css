.checkout-flex {
  display: flex;
  flex-direction: column;
}

.checkout-flex-1 {
  flex: 1;
  text-align: center;
  padding: 2rem;
  border-left: 5px solid #83364b;
  background: #fffdfd;
  /* margin: 2rem; */
}

#checkout > h1,
#checkout > h3 {
  font-weight: 900;
  text-align: center;
  color: #83364b;
}

#checkout > p {
  text-align: center;
  font-style: oblique;
}

.checkout-flex-2 {
  flex: 2;
}

.checkout-flex-2 > h2 {
  text-align: center;
}

#checkout-step-3-flex {
  display: flex;
  flex-direction: column;
  margin-bottom: 1rem;
}

#checkout-step-3-flex-2 {
  flex: 2;
  text-align: center;
}

#checkout-step-3-flex-1 {
  text-align: left;
}

#checkout-to-conf-button {
  padding: 1rem 4rem;
  background: #83364b;
  border: 1px solid #83364b;
  font-family: 'Poppins', sans-serif;
  color: #ffffff;
  font-weight: 900;
  transition: background 0.5s ease-in-out;
}

#order-summary-flex {
  display: flex;
  flex-direction: row;
}

.order-summary-flex-1 {
  flex: 1;
  text-align: left;
}

label {
  font-size: 0.8rem;
}

.input {
  padding: 0.5rem;
  background: transparent;
}

.shipping-option {
  display: inline;
}

.not-allowed {
  cursor: not-allowed;
}

p {
  font-weight: bold;
}

@media all and (min-width: 800px) {
  #checkout {
    padding: 5% 10%;
  }

  .checkout-flex {
    flex-direction: row;
    margin-left: 10%;
    margin-right: 10%;
  }

  .checkout-flex-1 {
    margin: 2rem;
  }

  #checkout-step-3-flex {
    flex-direction: row;
  }
}
