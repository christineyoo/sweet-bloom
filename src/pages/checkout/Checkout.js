import React, { Component } from 'react';
import AddressForm from '../../organisms/addressForm/AddressForm';
import './Checkout.css';
import ItemCard from '../../organisms/itemCard/ItemCard';
import { Link } from 'react-router-dom';
import ValidationError from '../../validationError/ValidationError';

class Checkout extends Component {
  state = {
    shipping: 10,
    name: {
      value: '',
      touched: false
    },
    cardNumber: {
      value: '',
      touched: false
    },
    expirationDate: {
      value: '',
      touched: false
    },
    cvc: {
      value: '',
      touched: false
    }
  };

  inputName = (name) => {
    this.setState({
      name: {
        value: name,
        touched: true
      }
    });
  };

  inputCardNumber = (cardNumber) => {
    this.setState({
      cardNumber: {
        value: cardNumber,
        touched: true
      }
    });
  };

  inputExpirationDate = (expirationDate) => {
    this.setState({
      expirationDate: {
        value: expirationDate,
        touched: true
      }
    });
  };

  inputCvc = (cvc) => {
    this.setState({
      cvc: {
        value: cvc,
        touched: true
      }
    });
  };

  validateName = () => {
    const name = this.state.name.value.trim();
    if (name.length === 0) {
      return 'A name is required.';
    }
  };

  validateCardNumber = () => {
    const cardNumber = this.state.cardNumber.value.trim();
    if (cardNumber.length === 0) {
      return 'A card number is required.';
    }
  };

  validateExpirationDate = () => {
    const expirationDate = this.state.expirationDate.value.trim();
    if (expirationDate.length === 0) {
      return 'An expiration date is required.';
    }
  };

  validateCvc = () => {
    const cvc = this.state.cvc.value.trim();
    if (cvc.length === 0) {
      return 'A CVC number is required.';
    }
  };

  renderItemCards = () => {
    const itemsInCart = this.props.itemsInCart;
    if (itemsInCart.length === 0) return null;
    return itemsInCart.map((item, i) => {
      return (
        <ItemCard
          key={i}
          isOnCheckout
          name={item.name}
          price={item.price}
          quantity={item.quantity}
          url={item.url}
        />
      );
    });
  };

  totalPrice = () => {
    const itemsInCart = this.props.itemsInCart;
    if (itemsInCart.length === 0) return null;
    const itemTotals = itemsInCart.map((item) => +item.price * +item.quantity);
    const totalPrice = itemTotals.reduce((res, curr) => res + curr);
    return totalPrice;
  };

  updateShipping = (e) => {
    this.setState({ shipping: e });
    this.props.handleShipping(e);
    return this.state.shipping;
  };

  render() {
    return (
      <div id='checkout'>
        <h1>Checkout</h1>
        <section className='checkout-flex'>
          <div className='checkout-flex-2'>
            <h2>Step 1 - Shipping Address</h2>
            <AddressForm />
            <h2>Step 2 - Payment Method</h2>
            <form className='form-flex'>
              <label htmlFor='name'>Name</label>
              <input
                id='name'
                name='name'
                type='text'
                onChange={(e) => this.inputName(e.target.value)}
                required
              />
              {this.state.name.touched && (
                <ValidationError message={this.validateName()} />
              )}
              <label htmlFor='card-number'>Card Number</label>
              <input
                id='card-number'
                name='card-number'
                type='text'
                onChange={(e) => this.inputCardNumber(e.target.value)}
                required
              />
              {this.state.cardNumber.touched && (
                <ValidationError message={this.validateCardNumber()} />
              )}
              <label htmlFor='exp-date'>Expiration Date</label>
              <input
                id='exp-date'
                name='exp-date'
                type='text'
                placeholder='mm/yy'
                onChange={(e) => this.inputExpirationDate(e.target.value)}
                required
              />
              {this.state.expirationDate.touched && (
                <ValidationError message={this.validateExpirationDate()} />
              )}
              <label htmlFor='cvc'>CVC</label>
              <input
                id='cvc'
                name='cvc'
                type='text'
                onChange={(e) => this.inputCvc(e.target.value)}
                required
              />
              {this.state.cvc.touched && (
                <ValidationError message={this.validateCvc()} />
              )}
            </form>
            <h3>Billing Address</h3>
            <AddressForm />
            <h2>Step 3 - Review Items and Shipping</h2>
            <div id='checkout-step-3-flex'>
              <section id='checkout-step-3-flex-2'>
                {this.renderItemCards()}
              </section>
              <section id='checkout-step-3-flex-1'>
                <h3>Delivery options</h3>
                <p>Today (2 hrs)</p>
                <input
                  type='radio'
                  name='option'
                  value={10}
                  onChange={(e) => this.updateShipping(e.target.value)}
                />
                <label>$10</label>

                <p>Tomorrow</p>
                <input
                  type='radio'
                  name='option'
                  value={7}
                  onChange={(e) => this.updateShipping(e.target.value)}
                />
                <label>$7</label>

                <p>Custom Date</p>
                <input
                  type='radio'
                  name='option'
                  value={5}
                  onChange={(e) => this.updateShipping(e.target.value)}
                />
                <label>$5</label>
              </section>
            </div>
          </div>
          <div className='checkout-flex-1'>
            <h2>Order Summary</h2>
            <div id='order-summary-flex'>
              <p className='order-summary-flex-1'>Items</p>
              <p className='order-summary-flex-1'>${this.totalPrice()}</p>
            </div>
            <div id='order-summary-flex'>
              <p className='order-summary-flex-1'>Shipping</p>
              <p className='order-summary-flex-1'>${this.state.shipping}</p>
            </div>
            <hr />
            <div id='order-summary-flex'>
              <p className='order-summary-flex-1'>
                <strong>Order Total</strong>
              </p>
              <p className='order-summary-flex-1'>
                <strong>${+this.totalPrice() + +this.state.shipping}</strong>
              </p>
            </div>
            <Link to={'/confirmation'}>
              <button>Place Order</button>
            </Link>
          </div>
        </section>
      </div>
    );
  }
}

export default Checkout;
