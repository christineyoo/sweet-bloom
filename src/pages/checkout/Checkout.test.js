import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Checkout from './Checkout';

it('renders Checkout UI as expected', () => {
  const mockedParams = {
    itemsInCart: []
  };
  const tree = renderer
    .create(
      <Router>
        <Checkout {...mockedParams} />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
