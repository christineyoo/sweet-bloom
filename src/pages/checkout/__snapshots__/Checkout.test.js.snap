// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Checkout UI as expected 1`] = `
<div
  id="checkout"
>
  <h1>
    Checkout
  </h1>
  <p>
    <i>
      Do not refresh the page.
    </i>
  </p>
  <section
    className="checkout-flex"
  >
    <div
      className="checkout-flex-2"
    >
      <h2>
        Step 1 - Shipping Address
      </h2>
      <form
        className="form-flex"
      >
        <label
          htmlFor="name"
        >
          Name
        </label>
        <input
          id="name"
          name="name"
          onChange={[Function]}
          required={true}
          type="text"
        />
        <label
          htmlFor="address"
        >
          Address
        </label>
        <input
          id="address"
          name="address"
          onChange={[Function]}
          required={true}
          type="text"
        />
        <label
          htmlFor="city"
        >
          City
        </label>
        <input
          id="city"
          name="city"
          onChange={[Function]}
          required={true}
          type="text"
        />
        <label
          htmlFor="state"
        >
          State
        </label>
        <input
          id="state"
          name="state"
          onChange={[Function]}
          required={true}
          type="text"
        />
        <label
          htmlFor="zipcode"
        >
          Zipcode
        </label>
        <input
          id="zipcode"
          name="zipcode"
          onChange={[Function]}
          required={true}
          type="text"
        />
      </form>
      <label
        id="verified-address"
      >
        <input
          onClick={[Function]}
          type="checkbox"
        />
        The address above is correct.
      </label>
      <h2>
        Step 2 - Payment Method
      </h2>
      <form
        className="form-flex"
      >
        <label
          htmlFor="name"
        >
          Name
        </label>
        <input
          id="name"
          name="name"
          onChange={[Function]}
          required={true}
          type="text"
        />
        <label
          htmlFor="card-number"
        >
          Card Number
        </label>
        <input
          id="card-number"
          name="card-number"
          onChange={[Function]}
          required={true}
          type="text"
        />
        <label
          htmlFor="exp-date"
        >
          Expiration Date
        </label>
        <input
          id="exp-date"
          name="exp-date"
          onChange={[Function]}
          placeholder="mm/yy"
          required={true}
          type="text"
        />
        <label
          htmlFor="cvc"
        >
          CVC
        </label>
        <input
          id="cvc"
          name="cvc"
          onChange={[Function]}
          required={true}
          type="text"
        />
      </form>
      <h2>
        Step 3 - Review Items
      </h2>
      <div
        id="checkout-step-3-flex"
      >
        <section
          id="checkout-step-3-flex-2"
        />
        <section
          id="checkout-step-3-flex-1"
        >
          <h3>
            Delivery options
          </h3>
          <p>
            Today (2 hrs)
          </p>
          <label>
            $10
          </label>
          <input
            name="option"
            onChange={[Function]}
            type="radio"
            value={10}
          />
          <p>
            Tomorrow
          </p>
          <label>
            $5
          </label>
          <input
            name="option"
            onChange={[Function]}
            type="radio"
            value={5}
          />
        </section>
      </div>
    </div>
    <div
      className="checkout-flex-1"
    >
      <h2>
        Order Summary
      </h2>
      <div
        id="order-summary-flex"
      >
        <p
          className="order-summary-flex-1"
        >
          Items
        </p>
        <p
          className="order-summary-flex-1"
        >
          $
        </p>
      </div>
      <div
        id="order-summary-flex"
      >
        <p
          className="order-summary-flex-1"
        >
          Shipping
        </p>
        <p
          className="order-summary-flex-1"
        >
          $
          10
        </p>
      </div>
      <hr />
      <div
        id="order-summary-flex"
      >
        <p
          className="order-summary-flex-1"
        >
          <strong>
            Order Total
          </strong>
        </p>
        <p
          className="order-summary-flex-1"
        >
          <strong>
            $
            10
          </strong>
        </p>
      </div>
      <button
        onClick={[Function]}
      >
        Checkout
      </button>
    </div>
  </section>
</div>
`;
